CodeEdit/Features/SourceControl/Client/GitClient+Push.swift

```diff
@@ -9,10 +9,13 @@ import Foundation
 
 extension GitClient {
     /// Push changes to remote
-    func pushToRemote(upstream: String? = nil) async throws {
+    func pushToRemote(remote: String? = nil, branch: String? = nil, setUpstream: Bool? = false ) async throws {
```

Comment:
I was trying to keep the usage clean by not requiring that every property needs to be defined for each usage.

Reply:
I actually mean that there's no reason to use an optional bool instead of a regular bool, since you're defaulting it to false. This means there is no scenario where someone would pass nil.

Reply:
Done